### PR TITLE
various compiler warnings

### PR DESCRIPTION
**Pull request by olafhering**

### CODE REVIEW

**Comment by mkortstiege** [comment]
Looks better now ;) Thanks. jenkins build and merge

File: xbmc/utils/FileUtils.cpp [hunk 1]
```diff
@@ -227,4 +227,5 @@ CDateTime CFileUtils::GetModificationDate(const std::string& strFileNameAndPath,
   {
     CLog::Log(LOGERROR, "%s unable to extract modification date for file (%s)", __FUNCTION__, strFileNameAndPath.c_str());
   }
+  return dateAdded;
```
Comment (xhaggi) [inline xbmc/utils/FileUtils.cpp]: we return `dateAdded` in line 224, please remove it.

**Comment by xhaggi** [comment]
meh to late :(

File: xbmc/utils/FileUtils.cpp (hunk 1 above)
Comment (olafhering) [inline xbmc/utils/FileUtils.cpp]: Is the compiler wrong? What happens after 'catch'?

Comment (ksooo) [inline xbmc/utils/FileUtils.cpp]: Compiler is right. If an exception is thrown, we land in the catch block, which does not return or (re)throw and therefore going after that block ... and then we would return 'nothing'.

Comment (xhaggi) [inline xbmc/utils/FileUtils.cpp]: It's a really minor. We return in 224 and now we also return at the end. it would be enough to have the return at end.

**Pull request merged**